### PR TITLE
feat(security)!: confirm push-mirror creation and filter the catalog by token scope

### PR DESCRIPTION
**Pull request by jmrplens**
## Description

Two hardening changes from the review of what is left of the September security audit. Both are breaking, which is the argument for doing them in a major rather than deferring them again.

**A push mirror can now only be created with confirmation.** `project.mirror_add` was mutating but not destructive, so nothing stood between a model persuaded by an issue body to "set up a backup mirror" and an entire repository being pushed to a host the attacker named. The audit called this the highest-value prompt-injection amplifier in the parameter-forwarding class. Nothing new was built for it: the spec is declared destructive, the validator already enforces that the spec and the route agree, and the confirm guard already runs off the route on all three surfaces. It stays a **create** spec rather than a delete one, so its idempotency hint stays false, which is the truth: two calls with one URL leave two mirror rows.

**The token-scope filter now filters the catalog instead of the registered names.** `scope_filter.go` documented its own gap in the source, and the gap was total on one surface: its keys are catalog group names, so nothing was ever removed on the individual surface and every admin tool stayed listed for a token with no `admin_mode`. It failed toward GitLab's own 403, so this was listing fidelity rather than authorization, but a model reading the listing concludes the capability exists. The filter now runs where the file said it should, before registration, which is also what [ADR-0014](docs/development/adr/adr-0014-catalog-first-runtime-architecture.md) says every surface is projected from.

What that changes, per surface:

| Surface | Change |
| --- | --- |
| individual | Everything: it removed nothing before. For a token without `admin_mode`, 104 actions on Free or CE and 126 on Ultimate |
| meta | Nothing new. The group was already dropped from the catalog, so the dispatcher was never registered and the deleted pass had nothing to match |
| dynamic | Nothing new, for the same reason |

The dead pass is deleted rather than left beside the new one. Its five keys are catalog group names, no tool registered outside the catalog has ever taken one, and both filtered surfaces removed the group before it could see one.

## Related Issue

No single issue: these are two of the items the review of `plan/security-audit-2026-09.md` ranked as belonging before 3.0.0.

## Type of Change

- [ ] Bug fix (non-breaking change that fixes an issue)
- [ ] New feature (non-breaking change that adds functionality)
- [ ] Enhancement (improvement to existing functionality)
- [x] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Refactoring (no functional change)
- [ ] Documentation update
- [ ] CI / build / tooling

## Changes Made

- Builds `project.mirror_add` with `toolutil.DestructiveAction`, so the confirm guard refuses the call before any request reaches GitLab.
- Records `mirror_add` in the known-non-keyword-destructive table, since its name carries no destructive verb.
- Applies the scope filter to the shared individual catalog after the operator's exclusions and before registration, and gives that cache key the scope component the meta key already had.
- Deletes `RemoveScopeFilteredTools` and its call.
- Rewrites the end-to-end scope-filter test, which was registering an unfiltered catalog and then removing names from the server, which is not what any surface does. It now builds the catalog through the assembler `cmd/server` calls.

## How to Test

1. `go test ./internal/tools/ ./internal/tools/projectmirrors/ ./cmd/server/ -count=1`.
2. Neuter the filter call and confirm `TestCreateServer_TokenScopeFilter_AppliesOnEverySurface/individual` fails with `gitlab_get_license is listed for a token with no admin_mode`. That is the regression the change exists to prevent, and it was verified to reproduce.
3. On the individual surface with a token lacking `admin_mode`, confirm `gitlab_get_license` is no longer listed.
4. Call `project.mirror_add` without `confirm` and confirm it is refused with no request reaching GitLab.

## Breaking Changes / Migration Notes

**Creating a push mirror now requires confirmation.** A client that supports MCP elicitation prompts and is otherwise unaffected. A client that cannot prompt must pass `confirm: true`: top level on `gitlab_execute_action`, inside `params` on `gitlab_project`, as an argument on `gitlab_add_project_mirror`. `GITLAB_MCP_YOLO_MODE` and `AUTOPILOT` skip it as they do for every other destructive action. The served input schema gains a `confirm` property, `x_destructive` becomes true and so does `destructiveHint`.

**The individual surface lists fewer tools for a narrow token.** Nothing that worked stops working: every action removed is one GitLab would have refused with a 403.

### Code Quality

- [x] Code compiles: `go build ./...`
- [x] Consolidated Go analysis passes on changed packages
- [x] Code is formatted: `make analyze-fix`
- [x] Follows idiomatic Go patterns and the conventions documented in `CLAUDE.md` / `.github/instructions/`

### Testing

- [x] All existing tests pass: `go test ./... -count=1`
- [x] New tests added for new functionality
- [x] `internal/tools` and `internal/tools/projectmirrors` are at 100% statement coverage, and both sides of the one condition added are driven
- [x] Edge cases and error scenarios covered
- [x] The end-to-end scope-filter test now drives the real assembler

### Documentation

- [x] Doc comments added for exported types/functions (godoc)
- [x] `docs/` updated: the group-granularity limit is recorded in `docs/concepts/security.md`
- [x] `README.md` and the generated artifacts regenerated
- [ ] If introducing/removing a tool: `docs/tools/{domain}.md` and tool counts updated
- [x] Commit messages follow [Conventional Commits](https://www.conventionalcommits.org/)

### Security

- [x] No secrets, tokens, or credentials in code, tests, fixtures, or logs
- [x] Input validation for user-provided parameters
- [x] Error messages do not leak sensitive information
- [x] No new dependencies with known vulnerabilities

### CODE REVIEW

File: docs/reference/tools/mirrors.md [hunk 1]
```diff
@@ -63,14 +63,16 @@ Retrieve the SSH public key for a specific remote mirror. This key is used for S
 
 ### `gitlab_add_project_mirror`
 
-Add a new remote mirror to a project. Specify the target URL and optionally configure direction, authentication method, and whether to sync only protected branches.
+Add a new remote mirror to a project. Specify the target URL and optionally configure authentication method and whether to sync only protected branches.
 
-| Annotation | **Create** |
-| ---------- | ---------- |
+| Annotation | **Create**, destructive |
+| ---------- | ----------------------- |
```
[the remark this answers is not in the document]
Comment (jmrplens) [reply docs/reference/tools/mirrors.md:73]: Right, and it is this change that made it wrong: `gitlab_add_project_mirror` is the first tool in the reference to be annotated with a shape other than **Delete** and destructive, so no page had contradicted its own legend before.

I took the sentence rather than the diff, because the two say slightly different things. Confirmation is not decided by the annotation reading "destructive"; it is decided by the **Destructive** column of the legend table, which the annotation carries for a shape that is not destructive by default. The legend now says that, names **Delete** as the shape that always carries it, and points at the entry three sections below as the exception, so a reader who follows the legend arrives at the right answer for this page rather than a generalisation that happens to hold today.

The legend table is deliberately untouched. It states the default flags of each shape, and those defaults are still correct: a **Create** is not destructive unless it is annotated so, which is exactly what this one is.

The failure mode you describe is the one that mattered here. A client that cannot prompt and follows the legend would omit `confirm: true` and be refused, which fails closed rather than open, but it fails on a call the page told it was safe to make, and that is the defect.
